sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -111,34 +144,40 @@ class LeastSquares(BaseLoss):
 
         loss(x_i) = 0.5 * (y_true_i - raw_pred_i)**2
 
-    This actually computes the half least squares loss to optimize simplify
+    This actually computes the half least squares loss to simplify
     the computation of the gradients and get a unit hessian (and be consistent
     with what is done in LightGBM).
     """
 
     hessians_are_constant = True
 
-    def __call__(self, y_true, raw_predictions, average=True):
+    def __call__(self, y_true, raw_predictions):
         # shape (1, n_samples) --> (n_samples,). reshape(-1) is more likely to
         # return a view.
         raw_predictions = raw_predictions.reshape(-1)
         loss = 0.5 * np.power(y_true - raw_predictions, 2)
-        return loss.mean() if average else loss
+        return loss
```

Comment:
Don't we want the loss of each sample to be dependent on the sample_weight?

Reply:
I was wondering about that too. The argument in my head against that, is that the loss of a single sample is independent of its weight in the dataset. It's only in the context of a dataset that the sample weight has a meaning.

Reply:
Good point.

On a related note, would it make more sense then to have `__call__` return the weighted average loss, and have a method `pointwise_loss()` (which would be the current `__call__`)?

Reply:
sure, but in that case we won't need the `pointwise_loss` cause I don't think we ever use it. We only care about the actual value of the loss in the context of the average loss for a dataset. Otherwise we only want to update the gradients and hessians. I think the only place where the `__call__` is used right now is in the tests, and nowhere else in the code.

Reply:
currently in master `__call__` is used for the early stopping scores in the code, and `__call__(average=False)` is used in `test_loss.py`.

So I would suggest that `__call__(average=False)` becomes `pointwise_loss()`, which would be used both in the test and in the new `__call__`, which would basically be `np.average(self.pointwise_loss(), weights=sample_weight)`.